src/subscription_manager/repolib.py

```diff
@@ -142,24 +142,41 @@ def enable_yum_plugins(cls):
                     (yum_plugin_file_name, err)
                 )
                 continue
+
             if len(result) == 0:
                 log.info('Configuration file of yum plugin: "%s" cannot be read' % yum_plugin_file_name)
                 continue
-            is_plugin_enabled = yum_plugin_config.getint('main', 'enabled')
-            if is_plugin_enabled == cls.YUM_PLUGIN_ENABLED:
+
+            is_plugin_enabled = False
+            if not yum_plugin_config.has_section('main'):
+                log.warn(
+                    'Configuration file of yum plugin: "%s" does not include main section. Adding main section.' %
+                    yum_plugin_file_name
+                )
+                yum_plugin_config.add_section('main')
+            elif yum_plugin_config.has_option('main', 'enabled'):
+                try:
+                    # Options 'enabled' can be 0 or 1
+                    is_plugin_enabled = yum_plugin_config.getint('main', 'enabled')
+                except ValueError:
+                    try:
+                        # Options 'enabled' can be also: true or false
+                        is_plugin_enabled = yum_plugin_config.getboolean('main', 'enabled')
+                    except ValueError:
+                        log.warn(
+                            "File %s has wrong value of options: 'enabled' in section: 'main' (not a int nor boolean)" %
+                            yum_plugin_file_name
+                        )
+
+            if is_plugin_enabled == cls.YUM_PLUGIN_ENABLED or is_plugin_enabled is True:
```

Comment:
Wouldn't 

```
is_plugin_enabled = yum_plugin_config.getint('main', 'enabled')
```

assign 1 or 0 to `is_plugin_enabled` and therefore the predicate `is_plugin_enabled is True` will always be false?

Reply:
When yum plugin configuration file contains following text:

    [main]
    enabled = true

then line 164 takes effect:

```python
is_plugin_enabled = yum_plugin_config.getboolean('main', 'enabled')
```

It assigns `True` or `False` to `is_plugin_enabled`. The `is_plugin_enabled is True` will not take any effect, because: `True == 1` returns `True` in Python. Thus you are right that it can be removed, but for different reason :-).